translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package collect_list
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestApplyTimestampFormatRule(t *testing.T) {
+	r := new(TimestampRegax)
+	var input interface{}
+	e := json.Unmarshal([]byte(`{
```

Comment:
Why are we unmarshaling JSON? Couldn't we just use a `map[string]interface{}`?

Reply:
I was following other translation rule tests for consistency.
https://github.com/aws/amazon-cloudwatch-agent/blob/70c071bb9093bd4f63a7630abb002f8660ea8341/translator/translate/logs/logs_collected/files/collect_list/ruleLogFilters_test.go#L19
https://github.com/aws/amazon-cloudwatch-agent/blob/70c071bb9093bd4f63a7630abb002f8660ea8341/translator/translate/logs/logs_collected/files/collect_list/ruleRetentionInDays_test.go#L32
https://github.com/aws/amazon-cloudwatch-agent/blob/70c071bb9093bd4f63a7630abb002f8660ea8341/translator/translate/logs/logs_collected/files/collect_list/collect_list_test.go#L24

Reply:
I have changed the `ruleTimestampFormat_test` to follow this pattern. Note: there is still a large chunck of unit tests using unmarshalling JSON, refactoring those is out of scope for this PR